src/preferences/listener.test.ts

```diff
@@ -70,4 +85,27 @@ describe("preferences/listener", function () {
 
     assert.ok(updatePreferencesStub.notCalled);
   });
+
+  for (const [previewSetting, previewContextValue] of [
+    [ENABLE_DIRECT_CONNECTIONS, contextValues.ContextValues.directConnectionsEnabled],
+    [ENABLE_PRODUCE_MESSAGES, contextValues.ContextValues.produceMessagesEnabled],
+  ]) {
+    for (const enabled of [true, false]) {
+      it(`should update the "${previewContextValue}" context value when the "${previewSetting}" setting is changed to ${enabled} (REMOVE ONCE PREVIEW SETTING IS NO LONGER USED)`, async () => {
```

Comment:
what would we do with this test if nothing is currently in preview?

Reply:
We'll update the test setup (or remove them entirely) if we don't have any preview settings